casper/src/main/scala/coop/rchain/casper/Casper.scala

```diff
@@ -554,4 +571,29 @@ sealed abstract class MultiParentCasperInstances {
         } yield ()
       }
     }
+
+  def fromConfig[
+      F[_]: Monad: Capture: NodeDiscovery: TransportLayer: Log: Time: ErrorHandler: SafetyOracle](
+      conf: CasperConf)(implicit scheduler: Scheduler, taskLog: Log[Task]): MultiParentCasper[F] = {
+    val genesis = Genesis
+      .fromBondsFile[Task](conf.bondsFile, conf.numValidators, conf.validatorsPath)
+      .unsafeRunSync
```

Comment:
Hmm throwing in an explicit Monad `Task` and calling `unsafeRunSync` feels against the paradigm. But lets deal with this latter.

Reply:
Yes, you're probably right. Although I'm not sure what else to do about it because I need an actual MultiParentCasper instance at the end, not wrapped in some other effect. So if I used and `F` in this function as we normally do then I would somehow need to unwrap it afterwards. I guess maybe I need to think about using a `Comonad`...

Reply:
@birchmd couldn't you `map` over the `Genesis.fromBondsFile[F]` returned value?